Tasks/Src/PetWatchdogTask.c

```diff
@@ -0,0 +1,84 @@
+/*-----------------------------------------------------------
+ * PET WATCHDOG TASK
+ - Attempts to pet watchdog within appropriate time interval
+ -----------------------------------------------------------*/
+#include "BPS_Tasks.h"
+#include "WDog.h"
+#include <timers.h>
+
+
+/**
+ * @brief Watchdog Event Group Initialization
+ */
+void Init_WDogEventGroup() {
+    // Event Group init
+    xWDogEventGroup_handle = xEventGroupCreateStatic( &xWDogEventGroup );
+    configASSERT( xWDogEventGroup_handle );          // check if handle is set 
+    xEventGroupClearBits(xWDogEventGroup_handle,     /* The event group being updated. */
+                         0xFF );                /* The bits being cleared. */
+}
+
+
+/**
+ * @brief Callback function for Watchdog Window Timer.
+ *        Sets event group bit to signal Watchdog.
+ */
+void WDog_WindowCallback(TimerHandle_t xTimer) {
+    xEventGroupSetBits(xWDogEventGroup_handle, TIMER_DONE);
+}
+
+/*--------------------------------------------------------*/
+
+void Task_PETWDOG() {
+    // Initialize GPIO output for Pins 5 (LED), 6 (Task 1), 7 (Task 2)
+    GPIO_InitTypeDef gpio_config = {
+        .Mode = GPIO_MODE_OUTPUT_PP,
+        .Pull = GPIO_NOPULL,
+        .Pin = GPIO_PIN_5 | GPIO_PIN_6 | GPIO_PIN_7
+    };
+
+    WDog_Init(gpio_config, WDog_Error_Handler);
+
+    /* RTOS Timer */
+    // vTimerResetState();
+    TimerHandle_t xWindowTimer;
+    StaticTimer_t xTimerBuffer;
+
+    xWindowTimer = xTimerCreateStatic( 
+                    "Timer",                /* Just a text name, not used by the RTOS kernel. */
+                    WDOG_WINDOW_MS,         /* The timer period in ticks????, must be greater than 0. */ 
```

Comment:
The comment says this is in ticks but the macro says it's in ms. Which is it?

Also, for reference, portMS_TO_TICKS exists for convenience when interacting with functions that take ticks as input.